lib/addons/vpc-cni/index.ts

```diff
@@ -1,15 +1,248 @@
+  */
+  enablePrefixDelegation?: boolean;
+  /**
+  * `WARM_ENI_TARGET` Environment Variable. Format integer.
+  * Specifies the number of free elastic network interfaces (and all of their 
+  * available IP addresses) that the ipamd daemon should attempt to keep 
+  * available for pod assignment on the node. 
+  */
+  warmEniTarget?: number;
+  /**
+  * `WARM_PREFIX_TARGET` Environment Variable. Format integer.
+  * Specifies the number of free IPv4(/28) prefixes that the ipamd daemon 
+  * should attempt to keep available for pod assignment on the node.
+  */
+  warmPrefixTarget?: number;
+  /**
+   * Secondary Subnet IDs for creating `ENIConfig`
+   */
+  customNetworkingConfig?: CustomNetworkingConfig;
+}
+
+
+export interface CustomNetworkingConfig {
+  /**
+   * Explicitly select individual subnets
+   * @default - Use all subnets in a selected group (all private subnets by default)
+   */
+  readonly subnets?: ISubnet[];
+}
+
+const defaultProps: CoreAddOnProps = {
+  addOnName: 'vpc-cni',
+  version: 'v1.12.1-eksbuild.2',
+  saName: 'vpc-cni',
+  namespace: 'kube-system',
+  controlPlaneAddOn: true,
+  configurationValues: {}
+};
+
+/**
+ * Implementation of VpcCni EKS add-on with Advanced Configurations.
  */
 export class VpcCniAddOn extends CoreAddOn {
+    
+  readonly vpcCniAddOnProps: VpcCniAddOnProps;
+  readonly id? : string;
+
+  constructor(props?: VpcCniAddOnProps) {
+    super({...defaultProps, ...props});
+    this.vpcCniAddOnProps = { ...defaultProps, ...props, };
+    (this.coreAddOnProps.configurationValues as any) = populateVpcCniConfigurationValues(props);
+  }
+
+  deploy(clusterInfo: ClusterInfo): Promise<Construct> {
+      const cluster = clusterInfo.cluster;
+      let clusterSecurityGroupId = cluster.clusterSecurityGroupId;
+      let doc: string;
 
-    constructor(version?: string) {
-        super({
-            addOnName: "vpc-cni",
-            version: version ?? "v1.12.0-eksbuild.1",
-            saName: "vpc-cni"
-        });
+      if ((this.vpcCniAddOnProps.customNetworkingConfig?.subnets)) {
+        for (let subnet of this.vpcCniAddOnProps.customNetworkingConfig.subnets) {
+          doc = readYamlDocument(__dirname + '/eniConfig.ytpl');
+          const manifest = doc.split("---").map(e => loadYaml(e));
+          const values: Values = {
+              availabilityZone: subnet.availabilityZone,
+              clusterSecurityGroupId: clusterSecurityGroupId,
+              subnetId: subnet.subnetId  
+          };
+          const manifestDeployment: ManifestDeployment = {
+            name: "EniCustomConfig" + subnet.availabilityZone,
+            namespace: this.coreAddOnProps.namespace!,
+            manifest,
+            values,
+          };
+          const kubectlProvider = new KubectlProvider(clusterInfo);
+          kubectlProvider.addManifest(manifestDeployment);
+        }
+      }
+      const addOnPromise = super.deploy(clusterInfo);
```

Comment:
should not we add dependency on the ENI config? Or vice versa - ENI config must go after vpc cni deployment?

Reply:
ENIConfig is not a direct dependency for Addon Deployment. Also ENICOnfig is in a loop of 3 manifest. So basically not required,